tests/workbenches/notebook-controller/test_kueue_notebook_spawning.py

```diff
@@ -0,0 +1,710 @@
+        assert notebook_pod_labels.get("kueue.x-k8s.io/queue-name") == LOCAL_QUEUE_NAME, (
+            "Pod should reference correct queue"
+        )
+
+        # Wait for the notebook pod to become ready
+        notebook_pod.wait_for_condition(
+            condition=Pod.Condition.READY, status=Pod.Condition.Status.TRUE, timeout=Timeout.TIMEOUT_5MIN
+        )
+
+        # Verify the pod is now running (not gated)
+        running_pods_final, gated_pods_final = check_gated_pods_and_running_pods(
+            labels=pod_labels, namespace=kueue_enabled_notebook_namespace.name, admin_client=admin_client
+        )
+
+        assert running_pods_final == 1, f"Expected exactly 1 running notebook pod, found {running_pods_final}"
+        assert gated_pods_final == 0, f"Expected no gated notebook pods, found {gated_pods_final}"
+
+        # ADDITIONAL CHECK: Verify Kueue is tracking exact resource usage
+        # This proves that Kueue is actually managing the workload, not just adding labels
+        LOGGER.info(f"Checking ClusterQueue {kueue_notebook_cluster_queue.name} resource usage...")
+        cluster_queue_usage = get_queue_resource_usage(
+            queue=kueue_notebook_cluster_queue, flavor_name=RESOURCE_FLAVOR_NAME
+        )
+        LOGGER.info(f"ClusterQueue usage: {cluster_queue_usage}")
+
+        LOGGER.info(f"Checking LocalQueue {kueue_notebook_local_queue.name} resource usage...")
+        local_queue_usage = get_queue_resource_usage(queue=kueue_notebook_local_queue, flavor_name=RESOURCE_FLAVOR_NAME)
+        LOGGER.info(f"LocalQueue usage: {local_queue_usage}")
+
+        # Determine expected resource usage based on total pod requests
+        # For normal_resources scenario:
+        # - Main notebook container: cpu="1", memory="1Gi"
+        # - OAuth proxy sidecar: cpu="100m", memory="64Mi"
+        # - Total: cpu="1100m", memory="1088Mi"
+        expected_cpu = "1100m"
+        expected_memory = "1088Mi"
+
+        # Verify that either ClusterQueue or LocalQueue shows the exact expected resource usage
+        cluster_tracks_exact = verify_queue_tracks_workload(
+            queue=kueue_notebook_cluster_queue,
+            flavor_name=RESOURCE_FLAVOR_NAME,
+            expected_cpu=expected_cpu,
+            expected_memory=expected_memory,
+        )
+        local_tracks_exact = verify_queue_tracks_workload(
+            queue=kueue_notebook_local_queue,
+            flavor_name=RESOURCE_FLAVOR_NAME,
+            expected_cpu=expected_cpu,
+            expected_memory=expected_memory,
+        )
+
+        assert cluster_tracks_exact or local_tracks_exact, (
+            f"Either ClusterQueue or LocalQueue should show exact resource usage matching notebook requests "
+            f"(CPU: {expected_cpu}, Memory: {expected_memory}). "
+            f"ClusterQueue usage: {cluster_queue_usage}, LocalQueue usage: {local_queue_usage}"
+        )
```

Comment:
Can you please use utilities? Also, the test seems to be too big, is it possible to split it?